File: packages/shared/src/types/preflight.ts
/**
 * Pre-flight check types — hardware/environment readiness.
 * @gharibo/shared
 */

/** Status of a single pre-flight check item. */
export type PreflightStatus = "READY" | "NOT_READY" | "UNKNOWN";

/** A single pre-flight check item. */
export interface PreflightCheckItem {
  /** Check identifier, e.g. "python", "torch", "cuda", "gpu", "vram". */
  check: string;
  status: PreflightStatus;
  /** Human-readable detail, e.g. "PyTorch 2.4.0" or "nvidia-smi not available". */
  detail: string;
}

/** Full pre-flight result returned by the trainer service. */
export interface PreflightResult {
  overallReady: boolean;
  items: PreflightCheckItem[];
  checkedAt: string;
}

File: services/trainer/routers/preflight.py
"""
Preflight router — GET /preflight returns real environment check results.
"""
from fastapi import APIRouter, Query
from core.preflight_checks import run_preflight

router = APIRouter()


@router.get("/preflight")
async def preflight(
    base_model: str = Query(None, alias="base_model"),
    dataset_id: str = Query(None, alias="dataset_id"),
):
    """Runs the real pre-flight check and returns results."""
    result = run_preflight(base_model=base_model)
    return {
        "overall_ready": result.overall_ready,
        "items": [
            {"check": item.check, "status": item.status, "detail": item.detail}
            for item in result.items
        ],
        "checked_at": result.checked_at,
    }
